classes/submission/SubmissionMetadataFormImplementation.php

```diff
@@ -1,59 +0,0 @@
-<?php
-
-/**
- * @file classes/submission/SubmissionMetadataFormImplementation.php
```

Comment:
This class is also used in QuickSubmit, so removing it will require a rewrite over there.

Reply:
Oh no. I see it uses `submissionMetadataFormTitleFields.tpl` and `submissionMetadataFormFields.tpl` too. As well as the `AuthorGridHandler`. All of this has been removed. I see two options: 1) move these files into QuickSubmit (but I have probably removed several locale strings that they use) or 2) rewrite QuickSubmit to use the new wizard as a base. Both seem like a lot of work.

Reply:
Or 3. Leave it in OJS/pkp-lib for now but mark it deprecated.

Reply:
Ok, I've restored the files related to QuickSubmit, but I'm slightly doubting the wisdom of this. Still, maybe pain later is better than pain now. I am probably vastly underestimating the work level, but it might not be terribly difficult to add the extra QuickSubmit fields on top of the new wizard, which would head off any future problems. I know Erik has QuickSubmit on his todo list for PS.

I've only restored the code for OJS. As far as I could tell, the QuickSubmit plugin doesn't support OMP or OPS. Is that right?

Some things to consider:

- There's a good chance that changes to the data models will not be reflected in the quick submit forms, and so we're likely to get divergence in the data structure. None of the validation layers built into the API are applied in QuickSubmit.
- I had to restore the author grid, which means I had to restore the old author forms. In addition to the data consistency problem, it means that a plugin like ROR will have to integrate with both sets of author forms for 3.4.
- Any third-party plugin developer who integrated with the old submission wizard or author forms will probably be confused when they go to update their plugins. The plugins won't effect the submission wizard, but all the hooks that they used before are still there to be grep'd in the code base. 

I isolated these changes in single commits in case we still want to reverse course. It required 5,000+ lines of code to be restored (many of them locale keys). You can see the restored code here:

- https://github.com/pkp/pkp-lib/pull/8349/commits/f21630e4df695c6d2702389d719b1859b52c3c78  
- https://github.com/pkp/ojs/pull/3579/commits/db25b70d913da783d69c08d2d7f0860fa5594464

I'm currently unable to do any tests, because (I think) the main branch is broken.

Finally, to restore the locale keys, I wrote a little script that would copy a locale key in all locales from another instance of OJS (in this case, the `main` branch). It worked like:

```
php lib/pkp/tools/restoreLocaleKeys.php "submission.submit.selectPrincipalContact" ~/pkp/ojs-main
```

Do you want me to add this tool to `main`? Not sure if this is something that's done often enough to bother.

Reply:
Oh, and just for record keeping in case we need it: here are all of the locale keys that I restored after restoring these forms:

- submission.submit.form.authorRequired
- submission.submit.form.titleRequired
- submission.submit.form.abstractRequired
- submission.submit.form.languagesRequired
- submission.submit.form.subjectsRequired
- submission.submit.form.disciplinesRequired
- submission.submit.form.keywordsRequired
- submission.submit.form.coverageRequired
- submission.submit.form.typeRequired
- submission.submit.form.sourceRequired
- submission.submit.form.rightsRequired
- submission.submit.form.citationsRequired
- submission.submit.form.wordCountAlert
- submission.submit.metadataForm.tip
- submission.type.tip
- submission.source.tip
- submission.rights.tip
- submission.submit.metadataForm.tip
- submission.submit.submissionLocaleDescription
- author.users.contributor.name
- author.users.contributor.email
- author.users.contributor.role
- author.users.contributor.includeInBrowse
- notification.addedAuthor
- notification.editedAuthor
- grid.action.editContributor
- submission.submit.form.contributorRoleRequired
- submission.submit.selectPrincipalContact